spec/support/features/session_helper.rb

```diff
@@ -162,7 +162,7 @@ def acknowledge_and_confirm_recovery_code
       click_on button_text, class: 'recovery-code-continue'
 
       code_words.size.times do |index|
-        fill_in "recovery-#{index}", with: code_words[index].downcase
+        fill_in "recovery-#{index}", with: code_words[index].downcase + 'xxxx'
```

Comment:
There's points to be made on either side about checking length before checking match. There's a timing attack vector where if we return failure too quickly, then the attacker can discover aspects of the system.

In any case, the minimum length is well-advertised. There's nothing to discover there. What, exactly, is the attack with submitting a too-short or too-long password, other than a DDoS, which we already have ways to mitigate?